<launch>
  <include file="$(find lecture_6_ex_t)/launch/include/vesc.launch"/>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/v4l/by-id/usb-HD_Camera_Manufacturer_USB_2.0_Camera-video-index0" />
    <param name="brightness" value="-1" />
    <param name="contrast" value="-1" />
    <param name="saturation" value="-1" />
    <param name="sharpness" value="-1" />

<!-- possible values: mmap, read, userptr -->
    <param name="io_method" value="mmap"/>

    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="framerate" value="30" />

<!-- possible values: yuyv, uyvy, mjpeg, yuvmono10, rgb24 -->
    <param name="pixel_format" value="yuyv" />

<!-- enable/disable autofocus -->
    <param name="autofocus" value="false" />
    <param name="focus" value="-1" />

<!-- enable/disable autoexposure -->
    <param name="autoexposure" value="false"/>
    <param name="exposure" value="47"/>

    <param name="gain" value="-1" />

<!-- enable/disable auto white balance temperature -->
    <param name="auto_white_balance" value="true" />
    <param name="white_balance" value="4000" />

    <param name="camera_frame_id" value="usb_cam" />
    <param name="camera_name" value="usb_cam" />
    <param name="camera_info_url" value="" />
  </node>

  <node pkg="lecture_6_ex_t" type="go_cam.py" name="go_cam" output="screen" />
</launch>
